projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/DockerRegistryService.java

```diff
@@ -34,6 +59,68 @@ public String builderImage() {
 
   // TODO: Implement
   public boolean dataJobImageExists(String imageName) {
+
+    if (registryType.equalsIgnoreCase("ecr")) {
+      return checkImageExistsInEcr(imageName);
+    }
+
+    return false;
+  }
+
+  private boolean checkImageExistsInEcr(String imageName) {
+    // imageName is a string of the sort:
+    // 850879199482.dkr.ecr.us-west-2.amazonaws.com/sc/dp/integration-test-6ebd50b6:b722432f9f0aaa43fcec6cb68993651145815990
+    String imageRepoTag = imageName.split("amazonaws.com/")[1];
+    ImageIdentifier imageIdentifier =
+        new ImageIdentifier().withImageTag(imageRepoTag.split(":")[1]);
+    String imageRepository = imageRepoTag.split(":")[0];
+
+    var credentials = awsCredentialsService.createTemporaryCredentials();
+
+    AmazonECR ecrClient;
```

Comment:
You can also move the creation of the AmazonECR client to a separate method. Likely something like `private AmazonECR createAmazonECRClient(AWSCredentialsService.Credentials credentials)`